reader.go

```diff
@@ -865,8 +864,20 @@ func (r *Reader) run() {
 		l.Printf("entering loop for consumer group, %v\n", r.config.GroupID)
 	})
 
+	attempt := 0
+	sleepSecs := time.Second
+
 	for {
+		if attempt >= 5 {
+			panic(fmt.Sprintf("unknown topic %v or partition %v", r.config.Topic, r.config.Partition))
+		}
+
 		if err := r.handshake(); err != nil {
+			if err == UnknownTopicOrPartition {
```

Comment:
Add a couple of comments mentioning the scenarios in which this happens.
1. When the topic doesn't exist, then `joinGroup` throws this error.
2. When more consumers are started than the number of partitions, then `syncGroup` throws this error.